java/flight/flight-sql/src/main/java/org/apache/arrow/flight/sql/FlightSQLUtils.java

```diff
@@ -0,0 +1,203 @@
+  private static final int BIT_WIDTH_64 = 64;
+  private static final boolean IS_SIGNED_FALSE = false;
+  private static final boolean IS_SIGNED_TRUE = true;
+
+  public static final ActionType FLIGHT_SQL_GETSQLINFO = new ActionType("GetSQLINFO",
+          "Retrieves details of SQL capabilities of the Flight server. \n" +
+                  "Request Message: N/A\n" +
+                  "Response Message: ActionGetSQLInfoResult");
+
+  public static final ActionType FLIGHT_SQL_GETCATALOGS = new ActionType("GetCatalogs",
+          "Retrieves a list of all catalogs available on the server. \n" +
+                  "Request Message: ActionGetCatalogsRequest\n" +
+                  "Response Message: ActionGetCatalogsResult");
+
+  public static final ActionType FLIGHT_SQL_GETSCHEMAS = new ActionType("GetSchemas",
+          "Retrieves a list of schemas available on the server. \n" +
+                  "Request Message: ActionGetSchemasRequest\n" +
+                  "Response Message: ActionGetSchemasResult");
+
+  public static final ActionType FLIGHT_SQL_GETTABLES = new ActionType("GetTables",
+          "Retrieves a list of tables available on the server. \n" +
+                  "Request Message: ActionGetTablesRequest\n" +
+                  "Response Message: ActionGetTablesResult");
+
+  public static final ActionType FLIGHT_SQL_GETTABLETYPES = new ActionType("GetTableTypes",
+          "Retrieves a list of table types available on the server. \n" +
+                  "Request Message: N/A\n" +
+                  "Response Message: ActionGetTableTypesResult");
+
+  public static final ActionType FLIGHT_SQL_GETPREPAREDSTATEMENT = new ActionType("GetPreparedStatement",
+          "Creates a reusable prepared statement resource on the server. \n" +
+                  "Request Message: ActionGetPreparedStatementRequest\n" +
+                  "Response Message: ActionGetPreparedStatementResult");
+
+  public static final ActionType FLIGHT_SQL_CLOSEPREPAREDSTATEMENT = new ActionType("ClosePreparedStatement",
+          "Closes a reusable prepared statement resource on the server. \n" +
+                  "Request Message: ActionClosePreparedStatementRequest\n" +
+                  "Response Message: N/A");
+
+  public static final List<ActionType> FLIGHT_SQL_ACTIONS = ImmutableList.of(
+          FLIGHT_SQL_GETSQLINFO,
+          FLIGHT_SQL_GETCATALOGS,
+          FLIGHT_SQL_GETSCHEMAS,
+          FLIGHT_SQL_GETTABLES,
+          FLIGHT_SQL_GETTABLETYPES,
+          FLIGHT_SQL_GETPREPAREDSTATEMENT,
+          FLIGHT_SQL_CLOSEPREPAREDSTATEMENT
+  );
+
+  /**
+   * Converts {@link java.sql.Types} values returned from JDBC Apis to Arrow types.
+   *
+   * @param jdbcDataType {@link java.sql.Types} value.
+   * @param precision    Precision of the type.
+   * @param scale        Scale of the type.
+   * @return The Arrow equivalent type.
+   */
+  public static ArrowType getArrowTypeFromJDBCType(int jdbcDataType, int precision, int scale) {
```

Comment:
this duplicates code in the jdbc to arrow package i believe, can we consolidate it?  I also thing there might be the need for custom mappings (we needed it JDBC conversion)